.github/workflows/merge_to_main.yml

```diff
@@ -0,0 +1,104 @@
+name: Merge to main
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  get-release-tag:
+    name: Publish linux artifacts into s3 bucket
+    runs-on: ubuntu-latest
+    outputs:
+      tag: "tmp-pr-${{ steps.get_pr_number.outputs.result }}"
+    steps:
+      - name: Get PR number
+        uses: actions/github-script@v6
+        id: get_pr_number
+        with:
+          script: |
+            return (
+              await github.rest.repos.listPullRequestsAssociatedWithCommit({
+                commit_sha: context.sha,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+              })
+            ).data[0].number;
+          result-encoding: string
+
+  publish-linux:
+    name: Publish linux packages to staging and production buckets
+    needs: get-release-tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Publish linux packages to staging
+        uses: newrelic/infrastructure-publish-action@v1.2.3
+        env:
+          TAG: ${{ needs.get-release-tag.outputs.tag }}
+          GPG_MAIL: ${{ secrets.LOGGING_GPG_MAIL }}
+          GPG_PASSPHRASE: ${{ secrets.OHAI_GPG_PASSPHRASE }}
+          GPG_PRIVATE_KEY_BASE64: ${{ secrets.OHAI_GPG_PRIVATE_KEY_BASE64 }} # base64 encoded
+          AWS_S3_BUCKET_NAME: "nr-downloads-ohai-staging"
+          AWS_S3_LOCK_BUCKET_NAME: "onhost-ci-lock-staging"
+          AWS_REGION: "us-east-1"
+          AWS_ACCESS_KEY_ID: ${{ secrets.OHAI_AWS_ACCESS_KEY_ID_STAGING }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.OHAI_AWS_SECRET_ACCESS_KEY_STAGING }}
+          AWS_ROLE_ARN: ${{ secrets.OHAI_AWS_ROLE_ARN_STAGING }}
+          AWS_ROLE_SESSION_NAME: ${{ secrets.OHAI_AWS_ROLE_SESSION_NAME_STAGING }}
+        with:
+          tag: ${{env.TAG}}
+          repo_name: "newrelic/fluent-bit-package"
+          schema: "custom"
+          schema_url: "https://github.com/newrelic/fluent-bit-package/releases/download/${{ env.TAG }}/generated-linux-schema.yaml"
+          aws_access_key_id: ${{ env.AWS_ACCESS_KEY_ID }}
+          aws_secret_access_key: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          aws_s3_bucket_name: ${{ env.AWS_S3_BUCKET_NAME }}
```

Comment:
Why is this (and many other properties) duplicated in the `env` and in the `with` blocks?

Reply:
This is how we used to do it in other workflows

Reply:
Can we probably simplify this and avoid duplication (either `with` or `env`)?